Tests/XCUITests/BaseTestCase.swift

```diff
@@ -113,6 +113,19 @@ class BaseTestCase: XCTestCase {
         waitFor(element, with: "exists == true", timeout: timeout, file: file, line: line)
     }
 
+    // is up to 25x more performant than the above waitForExistence method
+    func mozWaitForElementToExist(element: XCUIElement, timeoutInSeconds: TimeInterval) {
```

Comment:
Have you tried `mozWaitForElementtoExist` in other places? If it works, we could use this implementation for the current `waitForExistence`.

If this implementation only applies to the large number of rows, you may `override` this method jut for `DatabaseFixtureTest`. 🤔 💭

Reply:
I want to test a wholesale replacement of it but haven't gotten to it yet. want to give it a go? it will work for anytime any other `wait` strategy is used. or at least it should haha

Reply:
I haven't checked how this will work (or won't work) on the new xcode version. i should test that. thanks for the call out.